### PR TITLE
Split Warmer.warm() into smaller functions

### PR DESCRIPTION
**Pull request by 2opremio**
`Warmer.warming()` was a huge function (~250 loc), making it really hard to read.

This change splits the warmer into smaller functions, improving readability.

After this change `Warmer.warming()` is still 100 loc, but reads much better.

### CODE REVIEW

File: registry/cache/repocachemanager.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+			break updates
+		case fetchers <- struct{}{}:
+		}
+		awaitFetchers.Add(1)
+		go func() {
+			defer func() { awaitFetchers.Done(); <-fetchers }()
+			entry, err := c.updateImage(ctxc, registryClient, upCopy)
+			if err != nil {
+				if err, ok := errors.Cause(err).(net.Error); ok && err.Timeout() {
+					// This was due to a context timeout, don't bother logging
```
Comment (hiddeco) [inline registry/cache/repocachemanager.go]: Does this mean that in cases of repeating failure (e.g. the image registry is down), the daemon simply will not log anything? If so, I am not sure if this is desired (in cases of repeating failure).

Comment (2opremio) [reply registry/cache/repocachemanager.go]: I think you are right, but I think fixing that falls out of this refactoring.

Comment (hiddeco) [reply registry/cache/repocachemanager.go]: Sorry, I noticed later that this was moved code and not new code, I agree.

Comment (2opremio) [reply registry/cache/repocachemanager.go]: No need to apologize, let's fix that later.

File: registry/cache/repocachemanager.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+		imagesToUpdateMissingCount: missing,
+	}
+
+	return result, nil
+}
+
+// updateImages, refreshes the cache entries for the images passed. It may not succeed for all images.
+// It returns the values stored in cache, the number of images it succeeded for and the number
+// of images whose manifest wasn't found in the registry.
+func (c *repoCacheManager) updateImages(ctx context.Context, registryClient registry.Client, images []imageToUpdate) (map[string]image.Info, int, int) {
```
Comment (hiddeco) [inline registry/cache/repocachemanager.go]: Just a note; I am not a fan of the `.., int, int` return, but wrapping it in a struct to give it more context does not seem very meaningful either, the comment makes up for it a bit though.

Comment (2opremio) [reply registry/cache/repocachemanager.go]: Yeah, I am not too pleased with `fetchImagesResult` either

Happy to look into any suggestions you may have.

Maybe computing the containers outside of the functions? I gave that a short chance but it was hairy. I will give it another try.

Comment (2opremio) [reply registry/cache/repocachemanager.go]: I didn't find a cleaner way. At least not with a much bigger refactoring.

**Pull request merged**